### DISCUSSION TITLE
File Upload : Expose the choose callback to the #empty slot to enable clicking on the cloud icon in addition to the drag and drop

### DISCUSSION THREAD

**Discussion by hl037** [primefaces/community#2644] [PrimeVue]
I find it pretty confusing and frustrating for the user to see an "upload to cloud" icon that just... does nothing when clicked on...
It would be great to have the choose callback from the #header slot exposed to the #empty slot to implement a click event too...

**Comment by jsodeman** [comment]
Updating with the solution after I found this discussion via search and figured it out.

The documentation does not include it, but you can call .choose() on the ref of the FileUpload.

```html
<FileUpload ref="fileUploader" ... >
    <template #empty>
        <Button @click="selectFiles">
```

```vue
<script setup>
const fileUploader = ref(null);

function selectFiles() {
	fileUploader.value.choose();
}
```
